master/internal/api_notebook.go

```diff
@@ -80,3 +85,35 @@ func (a *apiServer) NotebookLogs(
 	// Keep the request context open until the actor stops.
 	return logStreamActor.AwaitTermination()
 }
+
+func (a *apiServer) LaunchNotebook(
+	ctx context.Context, req *apiv1.LaunchNotebookRequest,
+) (*apiv1.LaunchNotebookResponse, error) {
+	cmdParams, user, err := a.prepareLaunchParams(ctx, &protoCommandParams{
+		TemplateName: req.TemplateName,
+		Config:       req.Config,
+		Files:        req.Files,
+	})
+	if err != nil {
+		return nil, err
+	}
+
+	notebookLaunchReq := command.NotebookLaunchRequest{
+		CommandParams: cmdParams,
+		User:          user,
+	}
+	actorResp := a.m.system.AskAt(notebooksAddr, notebookLaunchReq)
+	if err = api.ProcessActorResponseError(&actorResp); err != nil {
+		return nil, err
+	}
+
+	notebookID := actorResp.Get().(resourcemanagers.TaskID)
+	actorResp = a.m.system.AskAt(notebooksAddr.Child(notebookID), &notebookv1.Notebook{})
```

Comment:
Calling `actorResp` something like `notebook` is probably more helpful to the reader. Similar comment elsewhere. Also thought on calling `NotebookLaunchRequest` just `LaunchNotebook`?

Reply:
re `notebook`

it's a kind of future for notebook. at the time when we assign it we don't know what it is and I  think it'd be if the name reflects that somehow. but as you mention it'd be good if it shows notebook in it somewhere too. it'll just become longer

Reply:
`notebookFut` is ok with me.

Reply:
(but it should be apparent its a future from the type, and go style usually is don't put types in names if you can help it)

Reply:
hmm some indication of the type in the name can help readability without you having the reader consult the typessytem. it's subjective though

Reply:
yah, i dont mind it, i think it's just one of those go recommended things.

Reply:
https://github.com/determined-ai/determined/pull/1454/commits/d8569d4a27e87fcc8e6e76e4dfdf0495f0fd6260